### PR TITLE
Start of elytron-security integration tests

### PR DESCRIPTION
**Pull request by starksm64**
Related to #1163 I have started an elytron-security extension integration set of tests focusing on the native image related issues to start. As part of this, I updated the Target_org_wildfly_security_x500_util_X500PrincipalUtil#asX500Principal(Principal, boolean) override to support the conversion functionality as it is used in the tests and verifies overriden method.

### CODE REVIEW

**Review by gsmet — APPROVED** [review]
I fixed a recent conflict. Looks good to me.

Waiting for CI again before merging.

**Comment by starksm64** [comment]
CI passed, merging and closing

**Pull request merged**
